doc/src/hugo/content/guides/authors/design/patterns/command-event/optional-parameters.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "3 - Optional Parameters"
+description: "Optional Version of Entity"
+date: 2022-02-25T10:50:32-07:00
+draft: "false"
+weight: 10
+---
+
+# Optional Versioning
+
+## Q: Should the event for each command have a version of the entity where all fields are optional,
+such that only changed data is sent back to gateway?
+
+### Example:
+
+type InfoUpdated {
+    name: String?,
+    address: Address?,
+    members: MemberId*,
+    isPrivate: Boolean?,
+}
+
+type ContactsUpdated is {
+    primaryContacts: MemberId+,
+    billingContacts: MemberId*,
+    distributionContacts: MemberId*
+}
+
+event OrganizationEstablished {
+    id: OrganizationId
+    info: Info,
+    contacts: Contacts,
+}
+
+event OrganizationInfoModified {
+    id: OrganizationId
+    info: InfoUpdated,
+}
+
+event UserContactsModified {
+    id: OrganizationId
+    contacts: ContactsUpdated,
+}
```

Comment:
The above three don't have optional fields for the payload, despite what it says in the question.